### PR TITLE
add initial code

### PR DESCRIPTION
**Pull request by lilic**
CodeBlocks only works for the GCC complier and xterm terminal as agreed.

https://phabricator.endlessm.com/T14001

### CODE REVIEW

File: com.CodeBlocks.App.json [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    "sha256": "30fb9d206df4ab3c2e2fec2451a0a7c9723bfed9a673cc9be71e881c529d5a03"
+                },
+                {
+                    "type": "shell",
+                    "commands": [ "autoreconf --warnings=all --verbose --install"]
+                }
+            ]
+        },
+        {
+            "name": "CodeBlocks metadata",
```
Comment (iaguis) [inline com.CodeBlocks.App.json]: flatpak-builder complains with

```
Module names like 'CodeBlocks metadata' containing ' ' are problematic. Expect errors.
```

I didn't see any errors though.

**Review by iaguis — CHANGES_REQUESTED** [review]
I get an assertion error on a fresh start:

image

If I choose to ignore it forever it doesn't happen again but I think we should fix it, can you have a look?

Otherwise it looks good.

File: com.CodeBlocks.App.json (hunk 1 above)
Comment (lilic) [reply com.CodeBlocks.App.json]: @iaguis will change it to just `"metadata"` then.

**Comment by lilic** [comment]
@iaguis can you have another look? thanks!

**Review by iaguis — APPROVED** [review]

**Comment by iaguis** [comment]
Seems to work fine now. LGTM.

**Pull request merged**